File: LeetCodeProblemsGo/LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex/LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex.go
package LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex

func maximumJumps(nums []int, target int) int {
	length := len(nums)
	dp := make([]int, length)
	for i := 0; i < length; i++ {
		dp[i] = -1
	}
	dp[0] = 0
	for i := 1; i < length; i++ {
		for j := 0; j < i; j++ {
			if dp[j] == -1 {
				continue
			}
			if (nums[i]-nums[j]) < target || (nums[i]-nums[j]) > target {
				continue
			}
			dp[i] = max(dp[i], dp[j]+1)
		}
	}
	return dp[length-1]
}

File: LeetCodeProblemsCMake/Problems/LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex.cpp
//
// Created by Saeed Jannati on 5/10/2026 A.
//

#include "LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex.h"

#include <math.h>
using namespace std;

int LeetCode2770MaximumNumberOfJumpsToReachTheLastIndex::maximumJumps(vector<int> &nums, int target) {
    int length = static_cast<int>(nums.size());
    vector<int> dp(length, -1);
    dp[0] = 0;
    for (int i = 1; i < length; i++) {
        for (int j = 0; j < i; j++) {
            if (dp[j] == -1)
                continue;
            if (abs(nums[i] - nums[j]) > target)
                continue;
            dp[i] = max(dp[i], dp[j] + 1);
        }
    }
    return dp[length - 1];
}
